File: UI/ecinema_mobile/lib/models/user_movie_list.dart
import 'movie.dart';

class UserMovieList {
  final int? id;
  final int? userId;
  final int? movieId;
  final String? listType; // 'watched', 'watchlist', 'favorites'
  final DateTime? createdAt;
  final bool? isDeleted;
  final Movie? movie;

  const UserMovieList({
    this.id,
    this.userId,
    this.movieId,
    this.listType,
    this.createdAt,
    this.isDeleted,
    this.movie,
  });

  factory UserMovieList.fromJson(Map<String, dynamic> json) {
    return UserMovieList(
      id: json['id'],
      userId: json['userId'],
      movieId: json['movieId'],
      listType: json['listType'],
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
      isDeleted: json['isDeleted'] ?? false,
      movie: json['movie'] != null ? Movie.fromJson(json['movie']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'userId': userId,
      'movieId': movieId,
      'listType': listType,
      'createdAt': createdAt?.toIso8601String(),
      'isDeleted': isDeleted,
      'movie': movie?.toJson(),
    };
  }
} 

File: eCinema/eCinema.Services/Database/Entities/UserMovieList.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eCinema.Services.Database.Entities
{
    public enum MovieListType
    {
        Watched,
        Watchlist,
        Favorites
    }

    public class UserMovieList
    {
        public int Id { get; set; }
        
        [Required]
        public int UserId { get; set; }
        
        [Required]
        public int MovieId { get; set; }
        
        [Required]
        [MaxLength(20)]
        public string ListType { get; set; } = string.Empty;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public bool IsDeleted { get; set; } = false;

        [ForeignKey(nameof(UserId))]
        public virtual User User { get; set; } = null!;
        
        [ForeignKey(nameof(MovieId))]
        public virtual Movie Movie { get; set; } = null!;

        [NotMapped]
        public MovieListType ListTypeEnum
        {
            get => Enum.TryParse<MovieListType>(ListType, true, out var result) ? result : MovieListType.Watchlist;
            set => ListType = value.ToString();
        }

        public static string GetListTypeString(MovieListType listType)
        {
            return listType.ToString().ToLower();
        }

        public static MovieListType GetListTypeFromString(string listType)
        {
            return Enum.TryParse<MovieListType>(listType, true, out var result) ? result : MovieListType.Watchlist;
        }
    }
}
